### PR TITLE
[C-3014] Add prompt on leaving upload form in progress

### PR DESCRIPTION
**Pull request by amendelsohn**
### Description
Adds a warning before navigating away if the form has any changed values.
Does not trigger on form submission or on the screen's back button.

*Note: Currently, this only blocks navigation within the app. Refreshing the page or navigating away via the address bar does not trip the condition.*

### Screenshots
Screenshot 2023-09-01 at 3 24 50 PM

### CODE REVIEW

**Review by dylanjeffers — COMMENTED** [review]
Cool! I think we'd want to run this by UX my guess is we should use the prompt we use when user tries to refresh the page when a block confirmation is still in flight

**Comment by amendelsohn** [comment]
> Cool! I think we'd want to run this by UX my guess is we should use the prompt we use when user tries to refresh the page when a block confirmation is still in flight

Ah, true. I'll see if I can use a version of that instead. Thanks!

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/amendel-upload-prompt

**Review by dylanjeffers — COMMENTED** [review]
@amendelsohn also can we just see what ux is thinking for this interaction in general? Like should we use the modal or the alert or something else?

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/amendel-upload-prompt

File: packages/web/src/components/navigation-prompt/NavigationPrompt.tsx [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+import { Location } from 'history'
+import { Prompt } from 'react-router-dom'
+
+import layoutStyles from 'components/layout/layout.module.css'
+import { Text } from 'components/typography'
+import { useNavigateToPage } from 'hooks/useNavigateToPage'
+
+import styles from './NavigationPrompt.module.css'
+
+interface Props {
```
Comment (dylanjeffers) [inline packages/web/src/components/navigation-prompt/NavigationPrompt.tsx:20]: Type and I usually call it componentnameprops

File: packages/web/src/pages/upload-page/forms/EditTrackForm.tsx [hunk 2]
```diff
@@ -107,11 +114,12 @@ export const EditTrackForm = (props: EditTrackFormProps) => {
 }
 
 const TrackEditForm = (props: FormikProps<TrackEditFormValues>) => {
-  const { values } = props
+  const { values, dirty } = props
   const isMultiTrack = values.trackMetadatas.length > 1
 
   return (
     <Form>
+      <NavigationPrompt when={dirty} messages={messages.navigationPrompt} />
```
Comment (dylanjeffers) [inline packages/web/src/pages/upload-page/forms/EditTrackForm.tsx:122]: Sickkkkk

**Review by dylanjeffers — APPROVED** [review]
Very cool great work

**Pull request merged**